### PR TITLE
feat: Improvements on `securitykey.validate`

### PR DESCRIPTION
**Pull request by phorward**
- `useSessionKey=True` is now default
- `pre_condition`-parameter to auto-raise a PreconditionFailed() exception
- Replaced and optimized calls in viur-core
- 100% backwards compatible to existing calls

I don't know why `useSessionKey=True` was always explicitly provided, as you can see below:
before

### CODE REVIEW

**Comment by phorward** [comment]
This can be seen as an improvement on #656

File: core/modules/file.py [hunk 1]
```diff
@@ -656,7 +655,7 @@ def add(self, skelType: SkelType, node=None, *args, **kwargs):
         if skelType == "leaf":  # We need to handle leafs separately here
             skey = kwargs.get("skey")
```
Comment (sveneberth) [inline core/modules/file.py:656]: If `kwargs` doesn't contains the `skey` it's `None`. But the type hint enforces a `str` (`def validate(key: str, ...)`).

Therefore I suggest this change:

```suggestion
            skey = kwargs.get("skey", "")
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Comment by sveneberth** [comment]
Let's talk about a decorator. I use this in ViUR2:

```
def skeyRequired(func=None, **decoratorKwArgs):
	"""Decorator that checks the skey before the method is called.

	Optional callable to pass keyword-arguments for the securitykey.validate-call.

	Example:
		>>> from server import exposed
		>>> @exposed
		>>> @skeyRequired
		>>> def yourMethod(self):
		>>>     return {"foo": "bar"}
	"""

	def outerWrapper(f):
		@functools.wraps(f)
		def wrapper(*args, **kwargs):
			if not securitykey.validate(kwargs.pop("skey", ""), **decoratorKwArgs):
				raise errors.PreconditionFailed("Missing or invalid skey")
			return f(*args, **kwargs)

		return wrapper

	if isinstance(func, (types.MethodType, types.FunctionType)):
		return outerWrapper(func)  # @skeyRequired
	else:
		return outerWrapper  # @skeyRequired() or @skeyRequired(**anyKwargs)
```

**Comment by phorward** [comment]
In today's viur-meeting, we decided to solve it by a decorator. I'm with you, but anyway would like to merge this PR as is and integrate the feature set into the code-base.

**Comment by phorward** [comment]
Closed as superseded by #764

**Pull request closed without merging**